@types/globals.d.ts

```diff
@@ -97,3 +97,5 @@ declare type Getter = () => any;
 declare type Setter = (value: any) => void;
 
 declare const Buffer: any;
+
+declare type EnumAlias<EnumT> = EnumT[keyof EnumT];
```

Comment:
It's kind of strange, why a normal enum can't be recognized by the editor ? Why don't we find a proper solution in the editor's interpretation ?

Reply:
Because the `export type PixelFormat = typeof PixelFormat;` yields a `typeof PixelFormat` format. The `const e: typeof PixelFormat` declares `e` as `PixelFormat` enumeration itself. So we can access its enumerators as `e.RGB8`.

In the other hand, `const e: PixelFormat`(That's `EnumAlias<typeof PixelFormat>` yields) declares `e` as a **value** of `PixelFormat`, ie, `string` or `number`.

For example:
```ts
function f(e: typeof PixelFormat) {}
f(PixelFormat.RGB8); // Error: PixelFormat.RGB8 is an enum value
f(PixelFormat); // Ok, but meaningless
```

Reply:
Will all those declarations be stripped out in the JS file